pkg/dockermanager/dockermanager.go

```diff
@@ -163,7 +163,7 @@ func (dm *Manager) Manage(
 			err := worker(cctx, handle)
 			if err != nil {
 				if errors.Is(err, context.Canceled) {
-					handle.S().Warnf("sidecar worker failed: %s", err)
+					handle.S().Debugf("sidecar worker failed: %s", err)
```

Comment:
This is an expected condition for every single `initContainer` within a testplan instance pod. Currently we have an `initContainer` which creates the `outputs` directory prior to the testplan binary starting, so that the loggers can store logs there.

Therefore I am downgrading to Debug, from Warn.

Reply:
We probably shouldn't even log this.

Reply:
Well, we can disable those log lines with the command flag for level of logs. This log line was still useful in confirming what's happening with initContainers that inherit the env vars from the pod.

Reply:
(we should at least note that this isn't a failure)

Reply:
Not sure what you mean. Do you mean renaming `sidecar worker: failed` to `sidecar worker: returned` ?

This is not a failure, and we don't log it as one. The problem is that the `initContainer` inherits the env vars, so sidecar `thinks` it should manage networks for it - something we don't really want or need.

The worker did return an error and it should, because its operation was canceled, I think. We could alternatively detect the cancelation of context in the worker and try not to return an error, but `nil`, but I am not sure how much worth is in doing that.

---

On a second though we might have a race here, if the `initContainer` completes in the time we get an event from Docker daemon and call `start()`, but the time we get to:

```
	info, err := container.Inspect(ctx)
	if err != nil {
		return nil, fmt.Errorf("inspect failed: %w", err)
	}

	if !info.State.Running {
		return nil, fmt.Errorf("not running")
	}
```

Then we could even get an error log line for the initContainer that it is `not running` anymore.

Reply:
Don't we return "context canceled" every time we shut down because the context was canceled (e.g., we got a stop event)?

Reply:
We do shut down by canceling the context, but we handle it differently at various places.

1. If get context canceled during the initial phase, we return it, and currently log it (the change in this PR, because this used to be something that "shouldn't happen")

2. If we cancel the context during the `network changes` for-loop (at `pkg/sidecar/sidecar.go`), we detect it and don't return it (we return `nil`).

Reply:
Ah...